CONTRIBUTING_CODE.md

```diff
@@ -0,0 +1,302 @@
+
+### Multiple remotes
+In order to keep your fork up-to-date with the main repository, one convenient
+way to do so is to set up [multiple remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)
+for `git`.
+For example, this way you can always keep the master branch of your forked repository
+up-to-date with the main repository and you can easily `git merge` recent changes into
+your feature branch. To set up the main ProbNum repository as a secondary remote run
+
+```shell
+$ cd probnum
+$ git remote add upstream https://github.com/probabilistic-numerics/probnum
+```
+
+Now in addition to `origin`, which is your fork of ProbNum, you have access to the main
+repository named `upstream`. Check your remotes are set up properly by executing the following command.
+
+```shell
+$ git remote -v
+origin	https://github.com/MyGithubAccount/probnum.git (fetch)
+origin	https://github.com/MyGithubAccount/probnum.git (push)
+upstream	https://github.com/probabilistic-numerics/probnum (fetch)
+upstream	https://github.com/probabilistic-numerics/probnum (push)
+```
+Fetch and checkout the master branch of ProbNum. 
+Since there exists already a local `master` branch tracking the master branch of your fork, we will name this one
+`probnum-master` which tracks the master branch of `upstream`.
+
+```shell
+$ git fetch upstream
+$ git checkout -b probnum-master upstream/master
+```
+
+### Code changes
+Next, create a new branch describing the feature you would like to implement. 
+```shell
+$ git checkout -b my-new-feature
+```
+You can now get started writing code in your new branch. 
+
+Commit your changes to this local branch. Make sure to keep the best practices regarding code quality in mind. 
+Next, set up a branch on your fork which is tracked by your local branch. 
+```shell
+git push --set-upstream origin my-new-feature
+```
+Once in a while, merge changes of the upstream master branch into your local `my-new-feature` branch.
+```shell
+$ git checkout probnum-master
+$ git pull
+$ git checkout my-new-feature
+$ git merge probnum-master
+```
+### Open a pull-request
+Once you are happy with your PR, make sure your branch is up-to-date with upstream `master`, and re-run all tests.
+Then, from your local branch, push the changes to your remote branch
+```shell
+$ git push
+```
+
+and do the PR via the github interface. There, briefly explain the changes as well as how they were tested. That's it!
+
+
+## Virtual environments
+
+Virtual environments (venvs) help you to get a separate, clean installation of ProbNum. 
+Each venv uses a specific Python version.
+Venvs are useful as you may need different Python versions and dependencies for different projects.
+So, first make sure that you are using the correct Python version, and also that your pip 
+installation is up to date.
+
+Install the [virtualenv](https://virtualenv.pypa.io/en/latest/) package.
+```shell
+$ pip install virtualenv
+```
+
+Go to the probnum root directory. 
```

Comment:
ProbNum

Reply:
I used probnum here because that is the directory name in case they follow the tutorial step by step, so it's a bit more intuitive in case someone doesn't know what that means. and I did not want to write: Go to the ProbNum root directory called `probnum`, as that sounded too weird.